internal/util/telemetry/reporter.go

```diff
@@ -83,16 +83,17 @@ type Reporter struct {
 
 // NewReporterOpts represents reporter options.
 type NewReporterOpts struct {
-	URL            string
-	Dir            string
-	F              *Flag
-	DNT            string
-	ExecName       string
-	P              *state.Provider
-	ConnMetrics    *connmetrics.ConnMetrics
-	L              *slog.Logger
-	UndecidedDelay time.Duration
-	ReportInterval time.Duration
+	URL              string
+	Dir              string
+	F                *Flag
+	DNT              string
+	ExecName         string
+	EnabledTelemetry *bool
```

Comment:
No, we don't need another parameter; there are enough of them already.

Reply:
ok, I will fix it in the next commit :)